libraries/Microsoft.Bot.Builder/TurnContext.cs

```diff
@@ -271,8 +271,6 @@ async Task<ResourceResponse[]> SendActivitiesThroughAdapter()
                 {
                     var activity = bufferedActivities[index];
 
-                    activity.Id = responses[index].Id;
```

Comment:
Isn't this required for logging? (I think this buffered activity is the one used in logging, and if we do not set this id then the logs will not have the correct activity.Id).

Also, I do not understand how removing this will fix a caching issue with the Prompt's activity.Id higher up.  This is a buffered activity, created from applying a ConversationReference from the activity being sent: https://github.com/microsoft/botbuilder-dotnet/blob/a1c89a31f9ade2fd8e8eb1288ddab86f9ede57d2/libraries/Microsoft.Bot.Builder/TurnContext.cs#L239

Reply:
Good shout Eric regarding the logging. I will test and confirm tomorrow when I am back (out today). If this is the case then I think we will need to simply send a copy of the RetryPrompt activity to fix the reported issues as we will need to keep the adapter as is.